unified-runtime/source/adapters/level_zero/v2/context.cpp

```diff
@@ -79,6 +79,8 @@ ur_result_t ur_context_handle_t_::release() {
   if (!RefCount.decrementAndTest())
     return UR_RESULT_SUCCESS;
 
+  asyncPool.cleanupPools();
```

Comment:
Please create a context destructor to put this in.

Or maybe, we can move this logic to the destructor of `ur_usm_pool_handle_t_`?

Reply:
Actually, it's not needed. `EnqueuedPool` destructor already handles the cleanup, and is called when context is destroyed.